### PR TITLE
🚨🚨🚨 Spring cleaning: PyTorch 1.10 🚨🚨🚨

### PR DESCRIPTION
**Pull request by muellerzr**
Removes a healthy chunk of code as we move to support a minimum of torch 1.10 from now on :)

(Set everyone to review so we're all aware!)

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
Thanks for the big clean up!

File: README.md [hunk 1]
```diff
@@ -224,7 +224,7 @@ If you like the simplicity of 🤗 Accelerate but would prefer a higher-level ab
 
 ## Installation
 
-This repository is tested on Python 3.7+ and PyTorch 1.4.0+
+This repository is tested on Python 3.7+ and PyTorch 1.10.0+
```
Comment (pacman100) [inline README.md]: Python version is 3.7 here. Python version PR needs to be merged first then followed by conflict resolution

File: src/accelerate/accelerator.py [hunk 2]
```diff
@@ -1335,7 +1327,7 @@ def prepare_model(self, model: torch.nn.Module, device_placement: bool = None, e
         if self.native_amp:
             model._original_forward = model.forward
             model_forward_func = model.forward.__func__ if hasattr(model.forward, "__func__") else model.forward
-            if self.mixed_precision == "fp16" and is_torch_version(">=", "1.10"):
+            if self.mixed_precision == "fp16":
```
Comment (pacman100) [inline src/accelerate/accelerator.py:1330]: the else part here becomes redundant?

**Review by pacman100 — COMMENTED** [review]
Thank you @muellerzr for all the cleaning wrt PyTorch version bump up! 🚀

Left a couple comments

File: README.md (hunk 1 above)
Comment (muellerzr) [reply README.md]: Done :)

File: src/accelerate/accelerator.py (hunk 2 above)
Comment (muellerzr) [reply src/accelerate/accelerator.py:1330]: Nice catch!

**Review by pacman100 — APPROVED** [review]

**Pull request merged**